shumai/io/network.ts

```diff
@@ -78,32 +80,47 @@ export async function tfetch(
   return null
 }
 
-export function connect(forward_url, backward_url) {
+export function connect(forward_url: string, backward_url: string) {
   const backward = async (grad) => {
     // possibly need to continue backward pass
     const jacobian = await tfetch(backward_url, grad.grad_in)
     if (jacobian) {
       await grad.in.backward(jacobian)
     }
   }
-  async function forward(t) {
-    return await tfetch(forward_url, t, { grad_fn: backward })
+  async function forward(t: sm.Tensor) {
+    return tfetch(forward_url, t, { grad_fn: backward })
   }
   return forward
 }
 
-export function serve(request_dict, options) {
+/* copy Bun's Serve Type, less fetch as we supply that logic */
+export type ServeOpts = {
+  port?: string | number
+  hostname?: string
+  baseURI?: string
+  maxRequestBodySize?: number
+  development?: boolean
+  error?: (
+    this: Server,
+    request: Errorlike
+  ) => Response | Promise<Response> | undefined | Promise<undefined>
+}
+
+/* TODO: specify a bettery type than any as it's a function */
+export function serve(request_dict: Record<string, any>, options: ServeOpts) {
   const user_data = {}
-  const statistics = {}
+  const statistics: Record<string, { hits: number; seconds: number }> = {}
   const sub_stat_fn = request_dict.statistics.bind({})
-  request_dict.statistics = async (u) => {
+  request_dict.statistics = async (u: sm.Tensor) => {
     if (sub_stat_fn) {
       const s = await sub_stat_fn(u)
-      return {...s, ...statistics }
+      return { ...s, ...statistics }
     }
-    return statistics
+    return { ...statistics }
```

Comment:
why expand here?

Reply:
Not seeing the value in this clone

Reply:
Yea, this was honestly a brain fart RE working on this a bit later than intended. Will fix in the next commit!